src/main/java/org/lantern/http/JettyLauncher.java

```diff
@@ -165,9 +168,11 @@ protected void doGet(HttpServletRequest req,
                 } else if (uri.startsWith("/proxy_google")) {
                     writeFileToResponse(resp, Proxifier.PROXY_GOOGLE);
                 } else {
-                    resp.addCookie(new Cookie("XSRF-TOKEN", model.getXsrfToken()));
-                    if (!resp.containsHeader("Content-Security-Policy")) {
-                        LanternUtils.addCSPHeader(resp);
+                    if (!LanternUtils.isDevMode() && !LanternUtils.isLanternPi()) {
+                        resp.addCookie(new Cookie("XSRF-TOKEN", model.getXsrfToken()));
+                        if (!resp.containsHeader("Content-Security-Policy")) {
+                            LanternUtils.addCSPHeader(resp);
+                        }
```

Comment:
Why not use the CSP policy for lantern-pi?  I imagine we can still do it if we just use the machine's IP address rather than "127.0.0.1" in our CSP header.